### PR TITLE
Add NATS Streaming doc to event notification doc

### PR DESCRIPTION
**Pull request by nitisht**
## Description
Added steps to configure and verify NATS Streaming messages with Minio server bucket notification

## Motivation and Context
Minio Server supports `NATS Streaming` but it is not available in docs. This PR adds steps to set up NATS Streaming as bucket notification target with Minio. 

## How Has This Been Tested?
Manually

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by harshavardhana — DISMISSED** [review]
LGTM tested streaming NATS locally.

File: docs/bucket/notifications/README.md [hunk 1]
```diff
@@ -562,6 +594,53 @@ go run nats.go
...
+// Import Go and NATS packages
+import (
+	"fmt"
+	"runtime"
+
+	"github.com/nats-io/go-nats-streaming"
+)
+
+func main() {
+	natsConnection, err := stan.Connect("test-cluster", "test-client")
```
Comment (vadmeste) [inline docs/bucket/notifications/README.md]: s/err/_/

File: docs/bucket/notifications/README.md [hunk 2]
```diff
@@ -562,6 +594,53 @@ go run nats.go
...
+	log.Println("Connected")
+
+	// Subscribe to subject
+	log.Printf("Subscribing to subject 'bucketevents'\n")
+	sub, _ := natsConnection.Subscribe("bucketevents", func(m *stan.Msg) {
```
Comment (vadmeste) [inline docs/bucket/notifications/README.md]: s/sub/_/

File: docs/bucket/notifications/README.md (hunk 1 above)
Comment (nitisht) [reply docs/bucket/notifications/README.md]: done

File: docs/bucket/notifications/README.md (hunk 2 above)
Comment (nitisht) [reply docs/bucket/notifications/README.md]: done

**Review by vadmeste — APPROVED** [review]
LGTM

**Review by harshavardhana — APPROVED** [review]

**Pull request merged**